### PR TITLE
Operate on a collection of (plugin) repositories

### PR DESCRIPTION
**Pull request by dblock**
### Description

With the goal to get to #22, use meta to operate on a collection of repositories. This allows us to do things like creating labels in many repos, creating issues, etc.
  
### Check List

- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
cc: @VachaShah and @CEHENKLE in case you need to do something on all the repos again

**Comment by dblock** [comment]
@saratvemulapalli What's the authoritative list of plugins that I should include here?

**Comment by saratvemulapalli** [comment]
> @saratvemulapalli What's the authoritative list of plugins that I should include here?

All the plugins which are public and are part of the opensearch-project should be included. 
If you need the exhaustive list, I would be happy to compile that for you.

**Comment by dblock** [comment]
I've added all the OpenSearch and OpenSerach Dashboards plugins in separate sub-projects. This is ready to review.

**Comment by dblock** [comment]
@saratvemulapalli

File: plugins/.gitignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/job-scheduler/
```
Comment (saratvemulapalli) [inline plugins/.gitignore:1]: Should we call this file `opensearch-plugins/.meta`? Just like dashboards file?

File: plugins/.meta [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "projects": {
+    "job-scheduler": "git@github.com:opensearch-project/job-scheduler.git",
+    "anomaly-detection": "git@github.com:opensearch-project/anomaly-detection.git",
+    "alerting": "git@github.com:opensearch-project/alerting.git",
+    "asynchronous-search": "git@github.com:opensearch-project/asynchronous-search.git",
+    "index-management": "git@github.com:opensearch-project/index-management.git",
+    "k-nn": "git@github.com:opensearch-project/k-nn.git",
+    "notebooks": "git@github.com:opensearch-project/dashboards-notebooks",
```
Comment (saratvemulapalli) [inline plugins/.meta:9]: May be I am missing something here, is this file intended to only have OpenSearch plugins and not Dashboards plugins?

Comment (dblock) [reply plugins/.meta:9]: Correct. I made 2 meta projects, `plugins/.meta` and `dashboards-plugins/.meta`, two sets of plugins.

File: plugins/.gitignore (hunk 1 above)
Comment (dblock) [reply plugins/.gitignore:1]: So either we include `opensearch` or we don't.

a) `plugins/.meta` and `dashboards-plugins/.meta` (here)
b) `opensearch-plugins/.meta` and `opensearch-dashboards-plugins/.meta`

Since this repo is already `opensearch-plugins` I thought that would be redundant.

Comment (dblock) [reply plugins/.gitignore:1]: WDYT @saratvemulapalli ?

File: plugins/.meta (hunk 2 above)
Comment (dblock) [reply plugins/.meta:9]: Note that some repos have both an opensearch plugin and a opensearch dashboards plugin, e.g.  `notebooks`.

Comment (saratvemulapalli) [reply plugins/.meta:9]: But dashboard plugins are part of `plugins/.meta` is what confused me.
Example: `"notebooks": "git@github.com:opensearch-project/dashboards-notebooks",`

File: plugins/.gitignore (hunk 1 above)
Comment (saratvemulapalli) [reply plugins/.gitignore:1]: That makes sense to me. Especially since the repo itself is called `opensearch-plugins`.
Lets exclude `opensearch`.

File: plugins/.meta (hunk 2 above)
Comment (dblock) [reply plugins/.meta:9]: That's because the OpenSearch plugin is in that repo, too: https://github.com/opensearch-project/dashboards-notebooks/tree/main/opensearch-notebooks. So the real question is whether we want 2 lists or 1, especially given that there will be duplication.

Comment (dblock) [reply plugins/.meta]: Because it's confusing and just easier to deal with repo names (rather than plugin names), I combined these and made the folder match the repo name.

**Review by CEHENKLE — APPROVED** [review]
It'd be interesting in the future to find ways to put related plugsin together (for example, if you want both Security BE and Security FE).
